### ISSUE TITLE
Switch over bool? is exhaustive in C# 8 (and newer) but not C# 7.3 (and older)

### ISSUE THREAD

**Issue by jhinder** [dotnet/roslyn#49280]
**Version Used**: VS 16.8 (3.8.0)

**Steps to Reproduce**:

```c#
int M(bool? p)
{
    switch (p)
    {
        case null: return -1;
        case true: return 1;
        case false: return 0;
    }
}
```

Set the language version to 8.0 or newer, then to 7.3 or older.

**Expected Behavior**: The switch is considered exhaustive under both language versions. No errors are emitted.

**Actual Behavior**: The code compiles under C# 8, but emits CS0616 under C# 7.3.
Adding a `default` arm suppresses the error, but leads to `code is unreachable` diagnostics by tools like Resharper.

C# 8:
![image](https://user-images.githubusercontent.com/823398/98795207-3f960000-240a-11eb-9873-6cc7356779a2.png)

C# 7.3:
![image](https://user-images.githubusercontent.com/823398/98795322-5e949200-240a-11eb-903d-61d0b515bf15.png)

**Comment by ufcpp** [comment]
see: https://github.com/dotnet/roslyn/issues/32806

This is for compatibility.

**Comment by jaredpar** [comment]
Closing as this is "by design" as @ufcpp noted.
